shell-script/public-gateway-checker.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+# IPFS public gateway checker v1.0 (shell script version)
+# original repository: https://github.com/ipfs/public-gateway-checker
+# this script added by Joss Brown: https://github.com/JayBrown
+
+JSON_URL="https://raw.githubusercontent.com/ipfs/public-gateway-checker/master/gateways.json"
```

Comment:
Isn't it better to hardcode the list?

Reply:
Better to have the list in one place instead of multiple ones though.

Reply:
We could (in a later version!) add some extra functionality. The script would then create an invisible file in `$HOME`, e.g. `.ipfg` (for "interplanetary gateways"), which it will read at every run, and diff it against the canonical list on GitHub. If it's empty, nothing happens, i.e. no diff, just default run. If there is content, it will diff: ignore doubles, don't ignore additional URLs, then run. There could also be an extra `--local` option, and then the script would *only* read gateway URLs from the local file. For now, though, grabbing it from a canonical list on github.com/ipfs is preferable.